tstest/integration/integration_test.go

```diff
@@ -287,6 +291,154 @@ func TestAddPingRequest(t *testing.T) {
 	t.Error("all ping attempts failed")
 }
 
+func TestTwoNodeConnectivity(t *testing.T) {
+	// t.Parallel()
+	bins := BuildTestBinaries(t)
+
+	env := newTestEnv(t, bins)
+	defer env.Close()
+
+	if err := tstest.WaitFor(20*time.Second, func() error {
+		// Create two nodes and hope that logs come out correctly
+		socks5Arg := "--socks5-server=" + "localhost:0"
+		n1 := newTestNode(t, env)
+		d1 := n1.StartDaemon(t, socks5Arg)
+		defer d1.Kill()
+
+		n2 := newTestNode(t, env)
+		d2 := n2.StartDaemon(t, socks5Arg)
+		defer d2.Kill()
+
+		n1.AwaitListening(t)
+		n2.AwaitListening(t)
+		n1.MustUp()
+		n2.MustUp()
+		n1.AwaitRunning(t)
+		n2.AwaitRunning(t)
+		// n1IP := n1.AwaitIP(t)
+		// n2IP := n2.AwaitIP(t)
+
+		defer func() {
+			d1.MustCleanShutdown(t)
+			d2.MustCleanShutdown(t)
+			d1.Kill()
+			d2.Kill()
+		}()
+
+		const sub = `SOCK-TEST : `
```

Comment:
rather than searching for substrings in the whole output, let's instead register a hook with the logcatcher that we can process per line.

I'll send a change for that part to make this sort of parsing easier in the future